### PR TITLE
Fix issues in bq to vcf

### PR DESCRIPTION
**Pull request by allieychen**
- Raise an error when there are no shards found.
- When parsing the genomic region filter, it should be case sensitive. For instance, chrY:1-1000, should have ref=chrY, rather than ref=chry, or the SELECT query will return the wrong results. This method is reused by partitioning, in which the matching of the reference should be case insensitive.

Issues: [447](https://github.com/googlegenomics/gcp-variant-transforms/issues/447),[446](https://github.com/googlegenomics/gcp-variant-transforms/issues/446)
Tested: unit tests

### CODE REVIEW

File: gcp_variant_transforms/libs/vcf_file_composer.py [hunk 1]
```diff
@@ -179,6 +179,8 @@ def _compose_blobs_to_one(self, blob_prefix):
     """
     blobs_to_be_composed = list(self._bucket.list_blobs(prefix=blob_prefix))
     logging.info('Total number of blobs is %d.', len(blobs_to_be_composed))
+    if not blobs_to_be_composed:
```
Comment (kemp-google) [inline gcp_variant_transforms/libs/vcf_file_composer.py]: This would have been a good candidate for a separate commit.  This is part of the value of keeping your commit messages complete from the start: you can then rebase and push when you have multiple unrelated changes in a single PR.

File: gcp_variant_transforms/libs/genomic_region_parser_test.py [hunk 2]
```diff
@@ -36,5 +36,9 @@ def test_parse_genomic_regions(self):
         genomic_region_parser.parse_genomic_region('chr'),
         ('chr', 0, genomic_region_parser._DEFAULT_END_POSITION)
     )
+    self.assertEqual(
```
Comment (kemp-google) [inline gcp_variant_transforms/libs/genomic_region_parser_test.py]: This could be folded into the test above (chr1 -> chrY)

File: gcp_variant_transforms/libs/vcf_file_composer.py (hunk 1 above)
Comment (allieychen) [reply gcp_variant_transforms/libs/vcf_file_composer.py]: I see! I will drop this PR and create a new one. :)
